tpp-app/src/onboarding/PeriodStart.js

```diff
@@ -1,42 +1,168 @@
+    if(range.startDate)
+      return (null);
+    else
+      return (<CalendarIcon style={styles.icon}/>);
+  }
 
-export default function PeriodStart ({ navigation }) {
   return (
-    <ImageBackground  source={OnboardingBackground} style={styles.container}>
-      <BackButtonContainer>
-        <BackButton title="" onPress={() => {navigation.navigate(STACK_SCREENS.PERIOD_LENGTH)}}/>
-      </BackButtonContainer>
-      <TitleText>
-        When did your {'\n'} period last start?
-      </TitleText>
-      <BodyText>
-        Record your last period or {'\n'} skip if you don’t know
-      </BodyText>
-
-      <TwoButtonContainer>
-        <SkipButton title="Skip" onPress={() => navigation.navigate(STACK_SCREENS.SYMPTOMS_CHOICES)}/>
-        <NextButton title="Next" onPress={() => 
-          {
-            PostInitialPeriodStart(new Date(2011, 11, 30));
-            navigation.navigate(STACK_SCREENS.SYMPTOMS_CHOICES);
-          }}/>
-      </TwoButtonContainer>
-    </ImageBackground>
+    <PaperProvider theme={theme}>
+      <ImageBackground source={OnboardingBackground} style={styles.container}>
+        <BackButtonContainer>
+          <BackButton title="" onPress={() => {navigation.navigate(STACK_SCREENS.PERIOD_LENGTH)}}/>
+        </BackButtonContainer>
+
+        <SafeAreaView pointerEvents="box-none" style={{ alignItems: 'center' }}>
+          <BackgroundShape style={{ top: 10 }}/>
+          <PeriodStartIcon width='130' height='130' style={{ bottom: "38%" }}/>
+          <BarIcon style={{ bottom: "30%" }}/>
+
+          <TitleText style={{ bottom: "28%" }}>
+            When did your {'\n'} period last start?
+          </TitleText>
+          <BodyText style={{ bottom: "28%" }}>
+            Record your last period or {'\n'} skip if you don’t know
+          </BodyText>
+
+          <InputContainer style={{ bottom: "25%" }}>
+            <DatePickerButton 
+              title={range.startDate ? range.startDate.toISOString().split('T')[0] : "Choose date"} 
+              onPress={() => setOpen(true)}
+              inputted={range.startDate}>
+            </DatePickerButton>
+            <CalendarIconPref/>
+          </InputContainer>
+          <DatePickerModal 
+            backgroundColor="#000000"
+            mode="range"
+            visible={open}
+            onDismiss={onDismiss}
+            startDate={range.startDate}
+            // calculates endDate if user entered periodLength; asks user to pick endDate manually otherwise
+            endDate={periodLength && range.startDate ? 
+                    new Date(range.startDate.getTime() + ((periodLength-1)*MILLISECPERDAY)) : range.endDate}
+            onConfirm={onConfirm}
+            onChange={onChange} 
+            validRange={{ 
+              // computes the latest possible endDate based on the value of periodLength
+              endDate: periodLength ? new Date(new Date().getTime() - ((periodLength-1)*MILLISECPERDAY)) : new Date() 
```

Comment:
I actually think this ```endDate``` should be the current date, not periodLength days before the current date, because the user could be on their first day of their period today. Sorry this was not explicitly discussed 😓 but in that case you would only record FLOW=MEDIUM for current dates. 

Ex: 
- today is March 28
- If user said periodLength=5 and periodStart=March 28, set only March 28 with FLOW=MEDIUM and don't set March 29-April 1

Ex 2: 
-today is March 28
- If user said periodLength=5 and periodStart=March 25, set only March 25-March 28 with FLOW=MEDIUM and don't set March 29

Reply:
@yichenesia ?